test/fail.test.js

```diff
@@ -1,23 +1,25 @@
-const {escape} = require('querystring');
-const test = require('ava');
-const nock = require('nock');
-const {stub} = require('sinon');
-const proxyquire = require('proxyquire');
-const SemanticReleaseError = require('@semantic-release/error');
-const {ISSUE_ID} = require('../lib/definitions/constants');
-const {authenticate} = require('./helpers/mock-github');
-const rateLimit = require('./helpers/rate-limit');
+import {escape} from 'node:querystring';
+
+import nock from 'nock';
+import quibble from "quibble"
+import SemanticReleaseError from '@semantic-release/error';
+import sinon from 'sinon';
+import test from 'ava';
+
+import {ISSUE_ID} from '../lib/definitions/constants.js';
+import {authenticate} from './helpers/mock-github.js';
+import * as RATE_LIMIT_MOCK from './helpers/rate-limit.js';
 
 /* eslint camelcase: ["error", {properties: "never"}] */
 
-const fail = proxyquire('../lib/fail', {
-  './get-client': proxyquire('../lib/get-client', {'./definitions/rate-limit': rateLimit}),
-});
+// mock rate limit imported via lib/get-client.js
+await quibble.esm('../lib/definitions/rate-limit.js', RATE_LIMIT_MOCK)
```

Comment:
would it be better for this to be inside the `beforeEach`?

Reply:
I don't think it's necessary, tests are passing just fine, but if you prefer I can move it into a a `test.before(() => {...})` block?